docs/source/index.rst

```diff
@@ -9,15 +13,32 @@ On a single machine dask increases the scale of comfortable data from
 *fits-in-memory* to *fits-on-disk* by intelligently streaming data from disk
 and by leveraging all the cores of a modern CPU.
 
-Users interact with dask either by making graphs directly or through the *dask
-collections* which provide larger-than-memory counterparts to existing popular
-libraries:
+Dask use cases
+==============
+
+* **Scientific computing libraries:** Projects like `xray`_ and `scikit-image`_ 
+  use dask.array for fast interactive analysis of large numeric datasets.
+* **Large unstructured or semi-structured data:** Dask.bag excels at 
+  processing large JSON blobs and log files.
+* **Scaling pandas-like data analysis to large datasets:** Dask.dataframe 
+  implements a part of the pandas interface, and can process data files larger than 
+  available memory.
+* **Algorithm development:** Working directly with dask graphs or dask.imperative 
+  is an excellent way to implement and test new algorithms in fields such as linear 
+  algebra, optimization, and machine learning.
```

Comment:
That sounds great to me! I'm with you 100% on the similarity to Dremels and the difficulty of briefly communicating a good understanding of the tool and its uses, given that there are far too many uses to enumerate each one. I don't have enough experience with dask to write such a section from the ground up, but I think what you've just written will fit very well.